src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -843,12 +845,23 @@ public function handlePlayerSkin(PlayerSkinPacket $packet) : bool{
 		$this->lastRequestedFullSkinId = $packet->skin->getFullSkinId();
 
 		$this->session->getLogger()->debug("Processing skin change request");
-		try{
-			$skin = $this->session->getTypeConverter()->getSkinAdapter()->fromSkinData($packet->skin);
-		}catch(InvalidSkinException $e){
-			throw PacketHandlingException::wrap($e, "Invalid skin in PlayerSkinPacket");
-		}
-		return $this->player->changeSkin($skin, $packet->newSkinName, $packet->oldSkinName);
+		$this->player->getServer()->getAsyncPool()->submitTask(new ProcessSkinTask(
+			$packet->skin,
+			function(?Skin $skin, ?string $error) use ($packet) : void{
```

Comment:
The player may not be connected anymore, or they may have changed to another different skin before the task finished